### PR TITLE
Sherry/check solutions

### PR DESCRIPTION
**Pull request by sliang100**
can I treat the board variable as a 2D array?

### CODE REVIEW

**Comment by brotatotes** [comment]
> can I treat the board variable as a 2D array?

I think currently the board is just a single array. It follows the entire board left to right, then top to bottom. So it has 81 elements. That means to access a certain `row` and `col` you have to multiply the `row` by 9 to move down past that many rows:

`board[9 * row + col]` would give you the element at `(col, row)`

File: src/App.js [hunk 1]
```diff
@@ -152,29 +154,146 @@ class Board extends React.Component {
...
+    let maxSudokuValue = 9;
+    const rowSet = new Set(); //create new set to store values of row
+    
+    //add values of board into Set
+    for (let col = 0; col < numCol; col++) {
+      rowSet.add(board[rowToVerify][col]);
+    }
+
+    //should have 9 values in row
+    if (rowSet.size != maxSudokuValue) {
```
Comment (brotatotes) [inline src/App.js:175]: what if there was a number in the set greater than 9 or less than 1? For example,

[0, 1, 2, 3, 4, 5, 6, 7, 8] has 9 elements but is not a valid row.

Comment (brotatotes) [reply src/App.js:175]: Nevermind, I see you check for that below.

**Comment by brotatotes** [comment]
You'll want to reopen this and then merge it into master, which will also automatically close the PR. If you just close it without merging, your changes won't be merged into master.

**Comment by RocketKevin** [comment]
rowSet.has(possibleSudoKuValue) always false. Something's wrong with .has().

File: src/App.js [hunk 2]
```diff
@@ -152,29 +154,154 @@ class Board extends React.Component {
...
+      return false;
+    }
+
+    // check if all possible values 1-9 are all in rowSet
+    for (let possibleSudoKuValue = 1; possibleSudoKuValue <= maxSudokuValue; possibleSudoKuValue++) {
+      if (rowSet.has(possibleSudoKuValue) === false) {
+        return false;
+      }
+    }
+    return true; //we've check that all numbers 1-9 are in the row
```
Comment (sliang100) [inline src/App.js:186]: I'm wondering if the === will make a difference? It looks like JS has autoboxing/auto unboxing, so I'm not sure why it always returns false

Comment (RocketKevin) [reply src/App.js:186]: No, rowSet.has(possibleSudoKuValue) is always false. If you log it, you'll see it

**Pull request merged**